src/plugins/data/public/search/search_interceptor/search_interceptor.ts

```diff
@@ -445,14 +452,72 @@ export class SearchInterceptor {
     if (this.bFetchDisabled) {
       const { executionContext, strategy, ...searchOptions } = this.getSerializableOptions(options);
       return this.deps.http
-        .post(`/internal/search/${strategy}${request.id ? `/${request.id}` : ''}`, {
-          version: '1',
-          signal: abortSignal,
-          context: executionContext,
-          body: JSON.stringify({
-            ...request,
-            ...searchOptions,
-          }),
+        .post<IKibanaSearchResponse>(
```

Comment:
You can probably change this type to `IKibanaSearchResponse | ErrorResponseBase`  `ErrorResponseBase` comes from elasticsearch client and seems to the the type of response that is returned by ES when you have an error. To test it you can created a custom DSL filter with a wrong syntax like
```
{
  "match_ph": {
    "bytes": 21212
  }
}
```

Reply:
it is very strange that we never considered using this type in our code. ES always return `200` even if there is an error in the query, it look like it just throw other status code if the call itself is wrong, not if the response is wrong.